### PR TITLE
chore: code style improve

### PR DESCRIPTION
**Pull request by yoyo837**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
ref #41317

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
@acyza

**Review by acyza — APPROVED** [review]
LGTM，Thank you

File: components/anchor/Anchor.tsx [hunk 1]
```diff
@@ -174,14 +174,12 @@ const AnchorContent: React.FC<InternalAnchorProps> = (props) => {
...
-      inkStyle.height = anchorDirection !== 'horizontal' ? `${linkNode.clientHeight}px` : '';
-      inkStyle.left = anchorDirection === 'horizontal' ? `${linkNode.offsetLeft}px` : '';
-      inkStyle.width = anchorDirection === 'horizontal' ? `${linkNode.clientWidth}px` : '';
-      if (anchorDirection === 'horizontal') {
+      const horizontalAnchor = anchorDirection === 'horizontal';
+      inkStyle.top = horizontalAnchor ? '' : `${linkNode.offsetTop + linkNode.clientHeight / 2}px`;
+      inkStyle.height = horizontalAnchor ? '' : `${linkNode.clientHeight}px`;
+      inkStyle.left = horizontalAnchor ? `${linkNode.offsetLeft}px` : '';
+      inkStyle.width = horizontalAnchor ? `${linkNode.clientWidth}px` : '';
+      if (horizontalAnchor) {
```
[the remark this answers is not in the document]
Comment (acyza) [reply components/anchor/Anchor.tsx:182]: 这里需要根据`anchorDirection`来确定设置位置和大小！

[the remark this answers is not in the document]
Comment (acyza) [reply components/anchor/Anchor.tsx:182]: 你可能是看错了位置，top和height的空字符串在`:`左边，left和width的空字符串在`:`右边。

[the remark this answers is not in the document]
Comment (kiner-tang) [reply components/anchor/Anchor.tsx:182]: 恩恩，刚刚 `top` 和 `height` 的条件看错了，就是感觉一堆三目运算看起来有点费劲

**Review by kiner-tang — APPROVED** [review]

**Pull request merged**